ui/src/main/java/com/midtrans/sdk/ui/views/creditcard/CreditCardDetailsFragment.java

```diff
@@ -0,0 +1,975 @@
+        boolean cardNumberValidity = checkCardNumberValidity();
+        boolean cardExpiryValidity = checkCardExpiryValidity();
+        boolean cardCVVValidity = checkCardCVVValidity();
+        return cardNumberValidity && cardExpiryValidity && cardCVVValidity;
+    }
+
+    private void setCardType() {
+        // Don't set card type when card number is empty
+        String cardNumberText = etCardNumber.getText().toString().trim();
+        if (TextUtils.isEmpty(cardNumberText) || cardNumberText.length() < 2) {
+            ivLogo.setImageResource(0);
+            return;
+        }
+
+        // Check card type before setting ivLogo
+        String cardType = CardUtilities.getCardType(cardNumberText);
+        switch (cardType) {
+            case CardUtilities.CARD_TYPE_VISA:
+                ivLogo.setImageResource(R.mipmap.ic_visa);
+                break;
+            case CardUtilities.CARD_TYPE_MASTERCARD:
+                ivLogo.setImageResource(R.mipmap.ic_mastercard);
+                break;
+            case CardUtilities.CARD_TYPE_JCB:
+                ivLogo.setImageResource(R.mipmap.ic_jcb);
+                break;
+            case CardUtilities.CARD_TYPE_AMEX:
+                ivLogo.setImageResource(R.mipmap.ic_amex);
+                break;
+        }
+    }
+
+    private void setBankType() {
+        // Don't set card type when card number is empty
+        String cardNumberText = etCardNumber.getText().toString().trim();
+        if (TextUtils.isEmpty(cardNumberText) || cardNumberText.length() < 7) {
+            ivBankLogo.setImageDrawable(null);
+            return;
+        }
+
+        String cleanCardNumber = cardNumberText.replace(" ", "").substring(0, 6);
+        String bank = presenter.getBankByBin(cleanCardNumber);
+
+        if (bank != null) {
+            switch (bank) {
+                case BankType.BCA:
+                    ivBankLogo.setImageResource(R.mipmap.ic_bank_bca);
+                    break;
+                case BankType.BNI:
+                    ivBankLogo.setImageResource(R.mipmap.ic_bank_bni);
+                    break;
+                case BankType.BRI:
+                    ivBankLogo.setImageResource(R.mipmap.ic_bank_bri);
+                    break;
+                case BankType.CIMB:
+                    ivBankLogo.setImageResource(R.mipmap.ic_bank_cimb);
+                    break;
+                case BankType.MANDIRI:
+                    ivBankLogo.setImageResource(R.mipmap.ic_bank_mandiri);
+                    break;
+                case BankType.MAYBANK:
+                    ivBankLogo.setImageResource(R.mipmap.ic_bank_mybank);
+                    break;
+                default:
+                    ivBankLogo.setImageDrawable(null);
+                    break;
+            }
+        } else {
+            ivBankLogo.setImageDrawable(null);
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    private void fadeIn() {
```

Comment:
Tara said we can remove the animation to prevent conflict with customised animation as in his (Xiaomi) phone